agent/lib/kontena/workers/image_pull_worker.rb

```diff
@@ -28,7 +28,8 @@ def pull_image(image, deploy_rev, creds)
       else
         info "pulling image with credentials: #{image}"
       end
-      update_image_cache("#{image}:#{deploy_rev}")
+      cache_key = "#{image}:#{deploy_rev}"
+      image_cache[cache_key] = Time.now.utc
```

Comment:
Setting the cache before pulling the image is still broken in the case of two concurrent instance deploys onto the same host node... there will be two `ServicePodWorker` actors running `ServicePods::Create` that both call into `ensure_image` with the same image and deploy_rev... while the first task sleeps, the second one will pick up the "cached" image and proceed to :boom: